streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValuesTest.java

```diff
@@ -138,6 +140,24 @@ public void close() { }
         assertArrayEquals(expected, supplier.theCapturedProcessor().processed.toArray());
     }
 
+    @Test
+    public void shouldEmitNoRecordIfTransformReturnsNull() {
+        final ProcessorContext context = mock(ProcessorContext.class);
+        final ValueTransformerWithKey<Integer, Integer, Integer> valueTransformer = mock(ValueTransformerWithKey.class);
+        final KStreamTransformValues.KStreamTransformValuesProcessor<Integer, Integer, Integer> processor = new KStreamTransformValues.KStreamTransformValuesProcessor<>(valueTransformer);
+        processor.init(context);
+        EasyMock.reset(context, valueTransformer);
+
+        final Integer inputKey = 1;
+        final Integer inputValue = 10;
+        EasyMock.expect(valueTransformer.transform(inputKey, inputValue)).andReturn(null);
+        EasyMock.replay(context, valueTransformer);
+
+        processor.process(inputKey, inputValue);
+
+        EasyMock.verify(context, valueTransformer);
```

Comment:
req: Since we do not need to validate `valueTransformer`, could you please remove it from the `verify()`.